crates/key-server/src/server.rs

```diff
@@ -155,23 +157,115 @@ impl Server {
             options,
+        })
+    }
+
+    /// Build the key_server_oid -> PoP HashMap for all server modes.
+    /// Fetches onchain for committee state and key server object ids.
+    async fn build_key_server_pop_map(
+        options: &KeyServerOptions,
+        master_keys: &MasterKeys,
+        mut grpc_client: SuiGrpcClient,
+    ) -> Result<HashMap<ObjectID, MasterKeyPOP>> {
+        match &options.server_mode {
+            ServerMode::Open {
+                key_server_object_id,
+            } => {
+                let key = master_keys
+                    .get_key_for_key_server(key_server_object_id)
+                    .expect("checked already");
+                let pop = create_proof_of_possession(key, &key_server_object_id.into_bytes());
+                Ok(HashMap::from([(*key_server_object_id, pop)]))
+            }
+            ServerMode::Permissioned { client_configs } => Ok(client_configs
+                .iter()
+                .filter(|c| {
+                    matches!(
+                        c.client_master_key,
+                        ClientKeyType::Derived { .. } | ClientKeyType::Imported { .. }
+                    )
+                })
+                .map(|c| {
+                    let key = master_keys
+                        .get_key_for_key_server(&c.key_server_object_id)
+                        .expect("checked already");
+                    let pop = create_proof_of_possession(key, &c.key_server_object_id.into_bytes());
+                    (c.key_server_object_id, pop)
+                })
+                .collect()),
+
+            ServerMode::Committee {
+                member_address,
+                state,
+            } => {
+                let committee_id = match state {
+                    CommitteeServerState::Active { committee_id } => committee_id,
+                    CommitteeServerState::Rotation {
+                        committee_id,
+                        next_committee_id: _,
+                    } => {
+                        // TODO: Handle next committee ID for rotation.
+                        committee_id
+                    }
+                };
+                info!(
+                    "Committee mode: {:?}, member address: {member_address}",
+                    state
+                );
+
+                let committee = fetch_committee_data(committee_id, &mut grpc_client).await?;
+                committee.is_finalized()?;
+
+                // Fetch PartialKeyServer info for this member
+                let (key_server_id, partial_key_server_info) =
+                    fetch_partial_key_server_info(&mut grpc_client, &committee, member_address)
+                        .await?;
+                info!(
+                    "Committee mode: KeyServer {} with party_id={}",
+                    key_server_id, partial_key_server_info.party_id
+                );
+
+                // Sign PoP over: key_server_id || party_id || partial_pk
```

Comment:
let's be consistent with the other call to create_proof_of_possession - the message already has pk (added in create_proof_of_possession)